scene/gui/tree.cpp

```diff
@@ -2449,7 +2452,7 @@ int Tree::draw_item(const Point2i &p_pos, const Point2 &p_draw_ofs, const Size2
 			int child_h = -1;
 			int child_self_height = 0;
 			if (htotal >= 0) {
-				child_h = draw_item(children_pos, p_draw_ofs, p_draw_size, c, &child_self_height);
+				child_h = draw_item(children_pos, p_draw_ofs, p_draw_size, c, child_self_height);
 				child_self_height += theme_cache.v_separation;
```

Comment:
This seems to be the only place where we care for the returned self height value. So I think this line adding vertical separation can be removed and instead we can assign `r_self_height` after we add this value to `label_h`.

Reply:
I'm not sure about whether to include the separator's height in the item's `self_height`. But I replaced the code with the following:

```c++
r_self_height = compute_item_height(p_item);
label_h = r_self_height + theme_cache.v_separation;
```

Reply:
It just seems redundant to add it in two places if these are the only two places that use this value. The name may be not the best, so I understand your hesitation. But I'd say this is pretty minor as far as semantics go.

Reply:
That said, I'm not insisting. We can merge it as is.